sources/subs/InlinePermissionsForm.class.php

```diff
@@ -0,0 +1,245 @@
+<?php
+
+/**
+ * Functions to support the permissions controller
+ *
+ * @name      ElkArte Forum
+ * @copyright ElkArte Forum contributors
+ * @license   BSD http://opensource.org/licenses/BSD-3-Clause
+ *
+ * @version 1.1 beta 2
+ */
+
+/**
+ * Class to initialize inline permissions sub-form and save its settings
+ */
+class Inline_Permissions_Form
+{
+	/**
+	 * @var string[]
+	 */
+	private $permissions = array();
+
+	/**
+	 * @var string[]
+	 */
+	private $illegal_permissions = array();
+
+	/**
+	 * @var string[]
+	 */
+	private $illegal_guest_permissions = array();
+
+	/**
+	 * @var int[]
+	 */
+	private $excluded_groups = array();
+
+	private $permissionsObject;
+	private $db;
+
+	/**
+	 * @return string[]
+	 */
+	public function getPermissions()
+	{
+		return $this->permissions;
+	}
+
+	/**
+	 * @param string[] $permissions
+	 */
+	public function setPermissions($permissions)
+	{
+		$this->permissions = $permissions;
+
+		// Load the permission list
+		$this->permissionList = array_map(
+			function ($permission)
+			{
+				return $permission[1];
+			}, $this->permissions
+		);
+
+		// Load the permission settings that guests cannot have
+		$this->illegal_guest_permissions = array_intersect(
+			array_map(
+				function ($permission)
+				{
+					return str_replace(array('_any', '_own'), '', $permission[1]);
+				}, $this->permissions
+			), $this->permissionsObject->getIllegalGuestPermissions()
+		);
+	}
+
+	/**
+	 * @return int[]
+	 */
+	public function getExcludedGroups()
+	{
+		return $this->excluded_groups;
+	}
+
+	/**
+	 * @param int[] $excluded_groups
+	 */
+	public function setExcludedGroups($excluded_groups)
+	{
+		$this->excluded_groups = $excluded_groups;
+	}
+
+	public function __construct()
+	{
+		$this->db = database();
+
+		// Make sure they can't do certain things,
+		// unless they have the right permissions.
+		$this->permissionsObject = new Permissions;
+		$this->illegal_permissions = $this->permissionsObject->getIllegalPermissions();
+	}
+
+	/**
+	 * Save the permissions of a form containing inline permissions.
+	 *
+	 * @param string[] $permissions
+	 */
+	public function save()
+	{
+		global $context;
+
+		$insertRows = array();
+		foreach ($this->permissionList as $permission)
+		{
+			if (!isset($_POST[$permission]))
```

Comment:
We have an HttpReq object hanging around that is usually preferred to $_POST. :angel: